Eatery Blue/UI/CompareMenus/CompareMenusTabsViewController.swift

```diff
@@ -119,6 +119,21 @@ class CompareMenusTabsViewController: UIViewController {
             make.width.equalToSuperview().inset(12)
             make.center.equalToSuperview()
         }
+
+        categoryViews.append(background)
+    }
+
+    func highlightFromScrollPercentage(_ percentage: Double) {
+        let index = Int((percentage * Double(categoryViews.count)).rounded(.toNearestOrAwayFromZero))
+        highlightCategoryAtIndex(index)
+    }
+
+    private func highlightCategoryAtIndex(_ index: Int) {
+        let boundedIndex = max(min(index, categoryViews.count - 1), 0)
+        for i in 0..<categoryViews.count {
+            let opacity: Float = i == boundedIndex ? 1 : 0.4
+            categoryViews[i].layer.opacity = opacity
+        }
```

Comment:
I recommend using a higher-order mapping function here instead of a for loop just for readability. Also since swift higher-order functions may be better optimized in this case.

You could do something like:
```
categoryViews.indices.map { index in
    // code here
}
```

Reply:
I used a for each instead